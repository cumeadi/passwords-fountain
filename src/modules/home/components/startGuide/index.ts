export { StartGuide } from './startGuide.component';
