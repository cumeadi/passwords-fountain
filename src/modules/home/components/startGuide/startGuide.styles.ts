import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { theme, media, stylelint, typography } from '@/common/theme';
setPragma(h);

export const GuideStep = styled('div')`
    color: ${theme.colors.darkBlue};
    ${typography.text16}
    margin-bottom: ${theme.spacing.m18};

    span {
        color: ${theme.colors.darkBlue};
    }

    ${media.gte(theme.breakpoints.s480)(stylelint.css`
        ${typography.text18}
        margin-bottom: ${theme.spacing.xl30};
    `)}
    
    ${media.gte(theme.breakpoints.m880)(stylelint.css`
        ${typography.text20}
    `)}
`;

export const Wrapper = styled('section')`
    border-radius: 25px;
    padding: ${theme.spacing.m18};
    background: ${theme.colors.pastelBlue};

    ${GuideStep}:last-of-type {
        margin-bottom: 0;
    }

    ${media.gte(theme.breakpoints.s480)(stylelint.css`
        padding: ${theme.spacing.xl30};
    `)}
`;
