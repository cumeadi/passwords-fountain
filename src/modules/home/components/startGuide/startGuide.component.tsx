import { h, VNode } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import { Wrapper, GuideStep } from './startGuide.styles';
import { Text } from '@/modules/localisation/components/text';

interface Step {
    id: number;
    textLabel: string;
    withMarkup?: boolean;
}

const steps: Step[] = [
    { id: 0, textLabel: 'startGuide.step1' },
    { id: 1, textLabel: 'startGuide.step2' },
    { id: 2, textLabel: 'startGuide.step3' },
    { id: 3, textLabel: 'startGuide.step4' },
    { id: 4, textLabel: 'startGuide.step5' },
];

export const StartGuide: TypedComponent<Props> = () => {
    const renderSteps = (): VNode[] =>
        steps.map(
            ({ id, textLabel, withMarkup = true }): VNode => (
                <GuideStep key={id}>
                    <Text withMarkup={withMarkup}>{textLabel}</Text>
                </GuideStep>
            )
        );

    return <Wrapper>{renderSteps()}</Wrapper>;
};

interface Props {}

StartGuide.propTypes = {};
