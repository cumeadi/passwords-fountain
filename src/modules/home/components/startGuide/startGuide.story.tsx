import { h, VNode } from 'preact';
import { StartGuide } from './startGuide.component';
import { withA11y } from '@storybook/addon-a11y';

export default {
    title: 'StartGuide',
    decorators: [withA11y],
};

export const defaultView = (): VNode => <StartGuide />;
