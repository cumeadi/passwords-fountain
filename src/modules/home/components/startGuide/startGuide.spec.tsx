import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { StartGuide } from './startGuide.component';

describe('StartGuide', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<StartGuide />);

        expect(asFragment()).toMatchSnapshot();
    });
});
