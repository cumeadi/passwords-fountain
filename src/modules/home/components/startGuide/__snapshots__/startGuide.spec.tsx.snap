// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartGuide should render correctly 1`] = `
<DocumentFragment>
  <section
    class="go2790461336"
  >
    <div
      class="go4191126316"
    >
      <span
        class="sanitized-translation"
      >
        Go to 
        <strong>
          <a
            href="https://fauna.com/"
          >
            https://fauna.com/
          </a>
        </strong>
         and create free account, login.
      </span>
    </div>
    <div
      class="go4191126316"
    >
      <span
        class="sanitized-translation"
      >
        Click 
        <strong>
          NEW DATABASE
        </strong>
         and use 
        <strong>
          PASSWORDS_FOUNTAIN
        </strong>
         as a database name.
        <br />
        This is important so please don't misspell it!
      </span>
    </div>
    <div
      class="go4191126316"
    >
      <span
        class="sanitized-translation"
      >
        Go to Settings -&gt; Account API Keys.
        <br />
         Now choose any name for your admin key and click 
        <strong>
          CREATE NEW ADMIN KEY.
        </strong>
      </span>
    </div>
    <div
      class="go4191126316"
    >
      <span
        class="sanitized-translation"
      >
        Your admin key will be shown only once, copy it. Don't show it to other people.
        <br />
         If you somehow lose admin key (you shouldn't) you can generate new one.
      </span>
    </div>
    <div
      class="go4191126316"
    >
      <span
        class="sanitized-translation"
      >
        Now you can connect our interface to your database.
        <br />
        Click on 
        <strong>
          App
        </strong>
         (upper-right corner) and start your journey!
      </span>
    </div>
  </section>
</DocumentFragment>
`;
